client/app/components/dashboards/widget.js

```diff
@@ -82,8 +82,13 @@ function DashboardWidgetCtrl($scope, $location, $uibModal, $window, $rootScope,
     EditParameterMappingsDialog.showModal({
       dashboard: this.dashboard,
       widget: this.widget,
-    }).result.then(() => {
+    }).result.then((valuesChanged) => {
       this.localParameters = null;
+
+      // refresh widget if any parameter value has been updated
+      if (valuesChanged) {
+        setTimeout(() => this.refresh(), 0); // must tick for refresh to work
```

Comment:
@kravets-levko do you know why this is needed here?

Reply:
@ranbena nope. @arikfr - do you know something about this?

Reply:
@ranbena (I removed my previous comment because I realized why you need it). You need to wait for Angular `$digest`  to apply new parameters. And you have to use `$timeout` service here (just replace `setTimeout` with it) - it will enforce `$digest` cycle before executing callback (`setTimeout` will not, and sometimes widget will not refresh as well)

Reply:
Done